cfme/tests/infrastructure/test_host.py

```diff
@@ -40,7 +55,38 @@ def navigate_and_select_quads(provider):
     return edit_view
 
 
+@pytest.mark.uncollectif(lambda provider: not provider.one_of(VMwareProvider))
+def test_discover_host(request, provider, appliance, hosts_ipaddresses):
+    if provider.delete_if_exists(cancel=False):
+        provider.wait_for_delete()
+
+    collection = appliance.collections.hosts
+
+    def _cleanup():
+        all_hosts = collection.all(provider)
+        if all_hosts:
+            collection.delete(*all_hosts)
+
+    _cleanup()
+    request.addfinalizer(_cleanup)
+
+    first_addr = provider.ip_address if hasattr(provider, 'ip_address') else hosts_ipaddresses[0]
+    last_addr = hosts_ipaddresses[-1]
+
+    collection.discover(first_addr, last_addr, esx=True)
+    hosts_view = navigate_to(collection, 'All')
+    expected_len = len(provider.data.get('hosts', {}))
```

Comment:
please move expected len to the below function

Reply:
The function below is being executed repeatedly by ``wait_for``. Moving ``expected_len`` to the function would mean I would need to execute this line of code up to 60x (``num_sec`` / ``delay``).